app/models/vm_scan.rb

```diff
@@ -600,6 +556,32 @@ def abort_retry(*args)
 
   private
 
+  def create_snapshot(vm)
```

Comment:
there are 2 returns in the method, both without parameter (thus `nil`) being passed back.  Again, I ask, how would the caller's `unless` work?